src/v/raft/append_entries_buffer.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "raft/types.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/gate.hh>
+
+namespace raft {
+
+class consensus;
+
+class append_entries_buffer {
```

Comment:
this class has no documentation of the internal state. 

what is the flow
what is the backpressure
what are potential downsides.